src/fsae_drivers/fsae_can.c

```diff
@@ -0,0 +1,241 @@
+                break;
+            case BMS_SSM_MODE_STANDBY:
+                bmsHeartbeat.state = CAN_BMS_STATE_STANDBY;
+                break;
+            case BMS_SSM_MODE_CHARGE:
+                bmsHeartbeat.state = CAN_BMS_STATE_CHARGE;
+                break;
+            case BMS_SSM_MODE_BALANCE:
+                bmsHeartbeat.state = CAN_BMS_STATE_BALANCE;
+                break;
+            case BMS_SSM_MODE_DISCHARGE:
+                bmsHeartbeat.state = CAN_BMS_STATE_DISCHARGE;
+                break;
+            default:
+                // You should never reach here
+                Board_Println("Unexpected curr_mode");
+                Error_Assert(ERROR_CONTROL_FLOW, bms_input->msTicks);
+                break;
+        }
+    }
+
+    // TODO when we get SOC working
+    bmsHeartbeat.soc = 0;
+
+    Can_Bms_Heartbeat_Write(&bmsHeartbeat);
+}
+
+void Send_Bms_Errors(void) {
+    Can_Bms_Error_T error_msg;
+    error_msg.type = get_error_status();
+    Can_Bms_Error_Write(&error_msg);
+}
+
+Can_Bms_ErrorID_T bms_error_to_can_error(ERROR_T error) {
+    switch (error) {
+        case ERROR_LTC6804_PEC:
+            return CAN_BMS_ERROR_LTC6804_PEC;
+        case ERROR_LTC6804_CVST:
+            return CAN_BMS_ERROR_LTC6804_CVST;
+        case ERROR_LTC6804_OWT:
+            return CAN_BMS_ERROR_LTC6804_OWT;
+        case ERROR_EEPROM:
+            return CAN_BMS_ERROR_EEPROM;
+        case ERROR_CELL_UNDER_VOLTAGE:
+            return CAN_BMS_ERROR_CELL_UNDER_VOLTAGE;
+        case ERROR_CELL_OVER_VOLTAGE:
+            return CAN_BMS_ERROR_CELL_OVER_VOLTAGE;
+        case ERROR_CELL_UNDER_TEMP:
+            return CAN_BMS_ERROR_CELL_UNDER_TEMP;
+        case ERROR_CELL_OVER_TEMP:
+            return CAN_BMS_ERROR_CELL_OVER_TEMP;
+        case ERROR_OVER_CURRENT:
+            return CAN_BMS_ERROR_OVER_CURRENT;
+        case ERROR_CAN:
+            return CAN_BMS_ERROR_CAN;
+        case ERROR_CONFLICTING_MODE_REQUESTS:
+            return CAN_BMS_ERROR_CONFLICTING_MODE_REQUESTS;
+        case ERROR_VCU_DEAD:
+            return CAN_BMS_ERROR_VCU_DEAD;
+        case ERROR_CONTROL_FLOW:
+            return CAN_BMS_ERROR_CONTROL_FLOW;
+        case ERROR_BRUSA:
+        case ERROR_NUM_ERRORS:
+            return CAN_BMS_ERROR_OTHER;
+        default:
+            // Unexpected error is control flow problem
+            return CAN_BMS_ERROR_CONTROL_FLOW;
+    }
+
+}
+
+/**
+ * @details Sends cell temperatures over Can
+ *
+ * @param pack_status datatype containing average cell temperature, maximum cell
+ * temperature, max cell temperature id, minimum cell temperature, and minimum
+ * cell temperature id
+ */
+void Send_Bms_CellTemps(BMS_PACK_STATUS_T * pack_status) {
+    Can_Bms_CellTemps_T cellTemps;
+    cellTemps.avg_cell_temp = pack_status->avg_cell_temp_dC;
+    cellTemps.min_cell_temp = pack_status->min_cell_temp_dC;
+    cellTemps.id_min_cell_temp = pack_status->min_cell_temp_position;
```

Comment:
Sorry for the late reply. I think we should bring it back to how it was at the beginning because this change isn't necessary. We have 6 modules, with 24 thremistors each (24*6 = 144 < 255). We should probably set MAX_NUM_MODULES to 6 because we have no reason to be using more than 6 modules.